### PR TITLE
Configure building Rust project for Raspberry Pi

### LINKED ISSUE #1: Set up pod operation programming environment
Having decided to use Rust for the pod operation program, we need to set up a Rust project with Cargo.
- Set up new Rust project with a `Cargo.toml` file
- Provide usage instructions via README
- Explore cross-compilation options to allow the compiled Rust program to be run on the Raspberry Pi

### PR DESCRIPTION
**Pull request by taesungh**
Resolves #1.

## Changes
- Add Cargo config to set build target platform and linker
- Include setup instructions in README

## Testing
1. Build the project with the ARM target from your personal machine
2. Attempt to run the executable on the Raspberry Pi in lab
3. Observe the program behaves as expected

## Remaining tasks
- [x] Include compiler installation instructions for macOS
- [x] Include compiler installation instructions for Windows
- [x] Include alternative instructions for using [cross](https://github.com/cross-rs/cross)
- [x] Update checks workflow to support cross-compilation

### CODE REVIEW

**Comment by taesungh** [comment]
Following from #15, we'll also need to update the checks workflow to be able to compile for arm-linux.

**Review by SM-dot — COMMENTED** [review]
Thanks for setting this up! 

- Is there a certain minimum required versions of Rust, cargo-watch, and Docker required?

- While the cross-compilation section is thorough for Windows/Linux, the instructions for MacOS appear relatively brief. Would it be possible to  provide additional details or addressing specific MacOS nuances to maintain consistency and completeness across all platforms?

**Comment by taesungh** [comment]
We specify Rust edition 2021 in `Cargo.toml`. We would expect developers to use modern versions of cargo and rustc although older versions _should_ also work. The Docker setup instructions actually apply to all host operating systems, and Sam will be providing a separate section for native compilation on Windows. There aren't many nuances with using the ARM compiler on macOS although we may need a separate script to distinguish host OS (pending discussion with Sam).

**Review by SM-dot — APPROVED** [review]

**Review by SM-dot — APPROVED** [review]

File: pod-operation/README.md [hunk 1]
```diff
@@ -4,20 +4,102 @@ This Rust package is for the main program to be run on the pod.
...
+brew tap messense/macos-cross-toolchains
+brew install armv7-unknown-linux-gnueabihf
+```
+
+#### Windows/Linux
+
+To cross-compile on Windows and Linux, a different compiler toolchain is needed. From the
+[Arm GNU Toolchain Downloads](https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads),
+download and install the **AArch32 GNU/Linux target with hard float (arm-none-linux-gnueabihf)**
+for Windows or Linux.
```
Comment (samderanova) [inline pod-operation/README.md:40]: A bit repetitive here, since we already state this in the sentence before.

**Review by samderanova — CHANGES_REQUESTED** [review]
Looks good, one small issue.

**Review by samderanova — APPROVED** [review]
Thanks for working on this!

**Pull request merged**